### PR TITLE
Allow to use MSTest new features

### PR DESCRIPTION
**Pull request by Evangelink**
Allow arcade users to rely on:

- `MSTest.Sdk`: current infra (before this change) always forces the installation of MSTest packages. This change allows to not install packages when user relies on `MSTest.Sdk` while still allowing to onboard to MSTest runner.

- MSTest meta package: This meta package introduced in 2.2.10 ensures that versions are aligned and provides the recommended default experience for our users.

- MSTest runner: Since 3.1, it's possible to use an alternative runner to VSTest.

### To double check:

* [ ] The right tests are in and the right validation has happened.  Guidance: https://github.com/dotnet/arcade/blob/main/Documentation/Validation.md

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets [hunk 1]
```diff
@@ -1,12 +1,26 @@
...
-  <ItemGroup>
+  <PropertyGroup>
+    <MSTestDependencyMode Condition=" '$(MSTestDependencyMode)' == '' ">Packages</MSTestDependencyMode>
+    <MSTestRunner Condition=" '$(MSTestRunner)' == '' ">VSTest</MSTestRunner>
+  </PropertyGroup>
+
+  <Target Name="_MSTestValidatePackageDeps" BeforeTargets="Build">
+    <Error Condition=" '$(UseMSTestMetaPackage)' == 'true' AND '$(UseMSTestSdk)' == 'true' " Text="Cannot use 'MSTest' metapackage and 'MSTest.Sdk' simultaneously." />
+  </Target>
+
```
Comment (Evangelink) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets:12]:
```suggestion
```

File: src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets [hunk 2]
```diff
@@ -1,12 +1,22 @@
 <!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. -->
 <Project>
 
-  <ItemGroup>
+  <PropertyGroup>
+    <MSTestDependencyMode Condition=" '$(MSTestDependencyMode)' == '' ">Packages</MSTestDependencyMode>
+    <MSTestRunner Condition=" '$(MSTestRunner)' == '' ">VSTest</MSTestRunner>
```
Comment (Evangelink) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets:6]: I don't know if you prefer to rely on multiple boolean properties and have some validation or to use some "enum" based properties instead. I went with this option here.

**Comment by Evangelink** [comment]
@ViktorHofer @mmitche Could I get a review please?

File: src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets (hunk 2 above)
Comment (ViktorHofer) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets:6]: I would probably go with a boolean which is easier to read.

**Review by ViktorHofer — DISMISSED** [review]

File: src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets (hunk 2 above)
Comment (Evangelink) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/MSTest/MSTest.targets:6]: Ok! I will update to use booleans and I will add the checks for invalid combination (that wasn't needed with the 
"enum").

**Comment by Evangelink** [comment]
@ViktorHofer Updated to use booleans instead of enum

**Review by mmitche — APPROVED** [review]

**Pull request merged**